spinn_front_end_common/interface/interface_functions/machine_bit_field_router_compressor.py

```diff
@@ -222,19 +222,25 @@ def __call__(
             bit_field_sorter_executable_path, threshold_percentage)
 
         # load and run binaries
-        system_control_logic.run_system_application(
-            compressor_executable_targets,
-            routing_table_compressor_app_id, transceiver,
-            provenance_file_path, executable_finder,
-            write_compressor_iobuf,
-            functools.partial(
-                self._check_bit_field_router_compressor_for_success,
-                host_chips=on_host_chips,
-                sorter_binary_path=bit_field_sorter_executable_path,
-                prov_data_items=prov_items),
-            [CPUState.FINISHED], True,
-            "bit_field_compressor_on_{}_{}_{}.txt",
-            [bit_field_sorter_executable_path], progress_bar)
+        try:
+            system_control_logic.run_system_application(
+                compressor_executable_targets,
+                routing_table_compressor_app_id, transceiver,
+                provenance_file_path, executable_finder,
+                write_compressor_iobuf,
+                functools.partial(
+                    self._check_bit_field_router_compressor_for_success,
+                    host_chips=on_host_chips,
+                    sorter_binary_path=bit_field_sorter_executable_path,
+                    prov_data_items=prov_items),
+                [CPUState.FINISHED], True,
+                "bit_field_compressor_on_{}_{}_{}.txt",
+                [bit_field_sorter_executable_path], progress_bar,
+                logger=logger)
+        except SpiNNManCoresNotInStateException as e:
```

Comment:
Well, I can add in just "Exception" since I have no idea what else might come out of there...  Note however that I do only want to catch that exception, the others can just raise...

Reply:
Hmm, from reading PEP8 and other advice, you really shouldn't catch exceptions that you don't want to handle...